From 13dbf05b1c40fee2f7448d7e0b4888f97d6a80e7 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:47 -0400
Subject: [cpufreq] Detect spurious invocations of update_policy_cpu()

Message-id: <1409235879-1283-74-git-send-email-prarit@redhat.com>
Patchwork-id: 89399
O-Subject: [RHEL7.1 PATCH BZ 1134639 073/185] cpufreq: Detect spurious invocations of update_policy_cpu()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 99ec899eafe2ec0a7dd96e9de5fa0a2bea3032ba
Author: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
Date:   Thu Sep 12 17:29:09 2013 +0530

    cpufreq: Detect spurious invocations of update_policy_cpu()

    The function update_policy_cpu() is expected to be called when the policy->cpu
    of a cpufreq policy is to be changed: ie., the new CPU nominated to become the
    policy->cpu is different from the old one.

    Print a warning if it is invoked with new_cpu == old_cpu, since such an
    invocation might hint at a faulty logic in the caller.

    Suggested-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index cedc3e4..8d6d774 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -969,7 +969,7 @@ static void cpufreq_policy_free(struct cpufreq_policy *policy)
 
 static void update_policy_cpu(struct cpufreq_policy *policy, unsigned int cpu)
 {
- if (cpu == policy->cpu)
+ if (WARN_ON(cpu == policy->cpu))
   return;
 
  /*
-- 
1.7.1